ctapipe/io/tableloader.py

```diff
@@ -0,0 +1,308 @@
+import re
+from typing import List, Union
+
+import numpy as np
+import tables
+from astropy.table import join, vstack, Table
+
+from ..core import Component, traits
+from ..instrument import SubarrayDescription, TelescopeDescription
+from .astropy_helpers import read_table
+
+__all__ = ["get_tel_ids", "get_structure", "TableLoader"]
+
+PARAMETERS_GROUP = "/dl1/event/telescope/parameters"
+IMAGES_GROUP = "/dl1/event/telescope/images"
+GEOMETRY_GROUP = "/dl2/event/subarray/geometry"
+TRIGGER_TABLE = "/dl1/event/subarray/trigger"
+SHOWER_TABLE = "/simulation/event/subarray/shower"
+TRUE_IMAGES_GROUP = "/simulation/event/telescope/images"
+
+by_id_RE = re.compile(r"tel_\d+")
+
+
+def get_tel_ids(
+    subarray: SubarrayDescription,
+    telescopes: List[Union[int, str, TelescopeDescription]]
+) -> List[int]:
+    ids = set()
+
```

Comment:
might want to also check if a string is given that its a valid TelescopeDescription string to ensure it's not a typo.  Right now, it will silently ignore any mis-spelled telescope strings.

Reply:
done